lib/plugins/threaded-signing/plugin.c

```diff
@@ -111,16 +118,26 @@ local_signature_info_create(const signature_info_t *signature_info)
 }
 
 /* Frees all allocated memory and resets members. Excluded are the worker thread members |thread|,
- * |mutex|, |cond| and |is_running|, but also the counter |nbr_of_unsigned_hashes|. */
+ * |mutex|, |cond| and |is_running|. */
 static void
 sv_threaded_plugin_reset(sv_threaded_plugin_t *self)
 {
   local_signature_info_free(self->signature_info);
   self->signature_info = NULL;
-  free(self->hash_to_sign);
-  self->hash_to_sign = NULL;
+
+  const int output_buffer_end = self->output_buffer_idx;
```

Comment:
- How about to remove line 128.
- Use "self->output_buffer_idx" on row 129 instead of "output_buffer_end".
- And then finally add "self->output_buffer_idx = 0" after the for loop.

Please check if this suggestion works.

Reply:
I think the code is easier to follow if we use output_buffer_end, but I'll change it if more people want me to.

Reply:
The name 'output_buffer_end' is actually wrong. It points to the first available element in the buffer and not the last one.
I think you should consider freeing all elements in the buffer instead of only the ones up to the "end". This will also make it possible for you to reuse allocated hashes. All these calloc and free are inefficient use of memory.
The input hashes are always the same length and the signatures are always at most 'max_signature_size'